src/runtime/agas/addressing_service.cpp

```diff
@@ -2843,30 +2843,35 @@ void addressing_service::send_refcnt_requests_non_blocking(
                 "addressing_service::send_refcnt_requests_non_blocking");
 #endif
 
-        // collect all requests for each locality
-        typedef std::map<naming::id_type, std::vector<request> > requests_type;
-        requests_type requests;
-
-        BOOST_FOREACH(refcnt_requests_type::const_reference e, *p)
+        // Only send decref requests if we aren't in shutdown mode, if we shut down, the corresponding
+        // component will get destroyed eventually.
+        if(!get_runtime().get_state() != runtime::state_shutdown)
```

Comment:
I believe, this has to be:

     if(!get_runtime().get_state() < runtime::state_shutdown)

instead.